src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.cs

```diff
@@ -228,40 +236,60 @@ internal abstract partial class AbstractAddImportCodeFixProvider<TSimpleNameSynt
-                if (assembly != null)
+                foreach (var reference in newReferences)
                 {
-                    findTasks.Add(finder.FindInMetadataAsync(project.Solution, assembly, reference, exact));
+                    var compilation = referenceToCompilation.GetOrAdd(reference, r => CreateCompilation(project, r));
+
+                    // Ignore netmodules.  First, they're incredibly esoteric and barely used.
+                    // Second, the SymbolFinder api doesn't even support searching them. 
+                    var assembly = compilation.GetAssemblyOrModuleSymbol(reference) as IAssemblySymbol;
+                    if (assembly != null)
+                    {
+                        findTasks.Add(finder.FindInMetadataAsync(project.Solution, assembly, reference, exact, linkedTokenSource.Token));
+                    }
                 }
+
+                await WaitForTasksAsync(allSymbolReferences, findTasks, nestedTokenSource, cancellationToken).ConfigureAwait(false);
             }
+        }
 
-            while (findTasks.Count > 0)
+        private async Task WaitForTasksAsync(
+            List<SymbolReference> allSymbolReferences,
+            HashSet<Task<List<SymbolReference>>> findTasks,
+            CancellationTokenSource nestedTokenSource,
+            CancellationToken cancellationToken)
+        {
+            try
             {
-                // Keep on looping through the 'find' tasks, processing each when they finish.
-                cancellationToken.ThrowIfCancellationRequested();
-                var doneTask = await Task.WhenAny(findTasks).ConfigureAwait(false);
+                while (findTasks.Count > 0)
+                {
+                    // Keep on looping through the 'find' tasks, processing each when they finish.
+                    cancellationToken.ThrowIfCancellationRequested();
+                    var doneTask = await Task.WhenAny(findTasks).ConfigureAwait(false);
 
-                // One of the tasks finished.  Remove it from the list we're waiting on.
-                findTasks.Remove(doneTask);
+                    // One of the tasks finished.  Remove it from the list we're waiting on.
+                    findTasks.Remove(doneTask);
 
-                // Add its results to the final result set we're keeping.
-                AddRange(allSymbolReferences, await doneTask.ConfigureAwait(false));
+                    // Add its results to the final result set we're keeping.
+                    AddRange(allSymbolReferences, await doneTask.ConfigureAwait(false));
 
-                // If we've got enough, no need to keep searching. 
-                // Note: We do not cancel the existing tasks that are still executing.  These tasks will
-                // cause our indices to be created if necessary.  And that's good for future searches which
-                // we will invariably perform.
```

Comment:
This comment was no longer valid.  Searching from Add-Import did not cause indices to be created.  Indices are created independently by the special incremental analyzer we have for that purpose.

So these extra tasks were literally wasting CPU.  Now we at least try to kill them off once we find enough project or metadata results.